lib/array_equals.rb

```diff
@@ -1,5 +1,26 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])
+    return true
+  elsif array1 == nil || array2 == nil
+    return false
+  end
+  
+  if array1.length == array2.length
```

Comment:
Style nit: using a "guarding if" that checks if `array1.length` **does not equal** `array2.length` is probably more readable here. (If nothing else, more programmers are used to that pattern - so it's easier for your coworkers to recognize.)